Objects/tupleobject.c

```diff
@@ -666,18 +666,24 @@ tuple_richcompare(PyObject *v, PyObject *w, int op)
     if (!PyTuple_Check(v) || !PyTuple_Check(w))
         Py_RETURN_NOTIMPLEMENTED;
 
+    if (v == w) {
+        Py_RETURN_RICHCOMPARE(0, 0, op);
+    }
+
```

Comment:
I'm aware of https://bugs.python.org/issue30907 but pyperformance was much less mature then, it was hard to go beyond microbenchmarks, and there was no broad quantifiable argument.